gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/catalog/GfxdDataDictionary.java

```diff
@@ -1636,15 +1636,15 @@ private void createGfxdSystemProcedures(TransactionController tc,
             DataTypeDescriptor.getCatalogType(Types.VARCHAR)};
         super.createSystemProcedureOrFunction("COMMIT_SNAPSHOT_TXID", sysUUID,
             argNames,argTypes, 0, 0, RoutineAliasInfo.NO_SQL, null, newlyCreatedRoutines,
-            tc, GFXD_SYS_PROC_CLASSNAME, true);
+            tc, GFXD_SYS_PROC_CLASSNAME, false);
       }
       {
         String[] argNames = new String[] { "txId"};
         TypeDescriptor[] argTypes = new TypeDescriptor[] {
             DataTypeDescriptor.getCatalogType(Types.VARCHAR)};
         super.createSystemProcedureOrFunction("USE_SNAPSHOT_TXID", sysUUID,
             argNames, argTypes, 0, 0, RoutineAliasInfo.NO_SQL, null,
-            newlyCreatedRoutines, tc, GFXD_SYS_PROC_CLASSNAME, true);
+            newlyCreatedRoutines, tc, GFXD_SYS_PROC_CLASSNAME, false);
       }
```

Comment:
@ashetkar We do need to think about this in future so track with an issue. All these special transaction procedures lack the requisite security i.e. anyone can commit/rollback any transaction. Ideally this should happen via thrift API or something so that it is tied to the specific client-side connection artifact and explicit TXId is never provided from this level. Please file a JIRA for 1.1 @suranjan